src/scikit_build_core/builder/get_requires.py

```diff
@@ -66,6 +67,25 @@ def _load_scikit_build_settings(
     return SettingsReader.from_file("pyproject.toml", config_settings).settings
 
 
+class RootPathResolver:
+    """Handle ``{root:uri}`` like formatting similar to ``hatchling``."""
+
+    def __init__(self, path: Path | None = None) -> None:
+        self.path = path if path else Path()
+
+    def __format__(self, fmt: str) -> str:
+        command, _, rest = fmt.partition(":")
+        if command == "parent":
+            parent = RootPathResolver(self.path.parent)
```

Comment:
Does this work if this is not an absolute path?


```python
>>> Path(".").parent
PosixPath('.')
```

Reply:
Oh good point, though it did pass the tests which used 1 parent call. My understanding is that `Path() == Path(".")`, but are they slightly nuanced w.r.t. `parent`.

Should probably resolve the default value to be safe, but it would require something other than `default_factory` there for that. Or we resolve it at each init.

Reply:
`default_factory=Path.cwd` would do it. But maybe we should always resolve, since `parent` will never go above `.`?

Reply:
Added resolve to `__post_init__` to be on the safe side. Apparently I had the same issue on the test side.